apache/superset
# chore(sqllab): Remove table metadata from state

### SUMMARY
Similar to https://github.com/apache/superset/pull/23488, https://github.com/apache/superset/pull/23257 and https://github.com/apache/superset/pull/24082, this commit migrates the table metadata(and extra metadata) state from redux to rdk-query to reduce the complexity of the sqlLab state.

### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF
After:

<img width="381" alt="Screenshot 2023-06-12 at 4 18 37 PM" src="https://github.com/apache/superset/assets/1392866/bb754707-9f80-4558-9a30-fd4b9baa5627">

Before:

<img width="525" alt="Screenshot 2023-06-12 at 5 37 10 PM" src="https://github.com/apache/superset/assets/1392866/b64cdd65-d5ee-4661-be38-7599f228c211">

### TESTING INSTRUCTIONS
Go to sqllab and select/remove table options from the left menu
check the table metadata including columns and indices

### ADDITIONAL INFORMATION
